OrbitBase/ThreadUtilsTest.cpp

```diff
@@ -3,10 +3,14 @@
 // found in the LICENSE file.
 
 #include <absl/synchronization/mutex.h>
-#include <gmock/gmock-matchers.h>
 #include <gtest/gtest.h>
 
-#include <chrono>
+#ifndef _WIN32
```

Comment:
Hmmm. This file has two flavors of that:

1. Line 19 and line 72:
```
#if _WIN32
#else
#endif
```

2. Line 54:
```
#ifdef __linux
#endif
```

Of course, this should be cleaned up and made consistent. But I guess not in this PR.

My thoughts on this:
1. `#if _WIN32` is wrong and should definitely be changed to `#ifdef _WIN32` (or something else).
2. I don't like having two separate branches: `#ifdef _WIN32 ... #endif #ifdef __linux ... #endif`. In that case the code might compile when neither macro is set which is a confusing silent failure. This is especially important since every included header could potentially change an existing macro. Silent bugs can be minimized by using fewer macros and fail if something is unexpected.
3. The header that I include here is technically not a linux-only header, rather a Unix/Posix header. So they would work on any other Unix-based OS like BSD or OSX. That's why I like `#ifndef _WIN32` here. In addition we mostly use `_WIN32` in the codebase (A quick search resulted in 56 usage of _WIN32/WIN32 and 8 usages of __linux/__linux__).
4. It might be confusing that `#ifndef _WIN32` means Linux. I guess that's also the reason why you added that comment. And that might be a strong argument for not doing it.

What are your thoughts on this?

Reply:
I think I share all of your arguments/thoughts.
I was not aware of point 3. In this case I'd guess I'd go for `#ifdef _WIN32 ... #else ... #endifdef`. I would hope that in this case it is clearer that "Not Windows" means Linux in our case. Further, I'd try to group those together.
This file was the most extreme example, so it would be nice, if you can address this for this file (and maybe the second one, I've commented on).

It doesn't need to be perfect, as it is already a very big step. But it would be cool to see this here a bit more consistent.

Reply:
Sure, makes sense. I also like the idea of having a consistent if-else-construct.

Reply:
Done.